test/unittest/dds/xtypes/serializers/idl/types/all_types.hpp

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * @file all_types.hpp
+ */
+
+#include <string>
+
+#include <fastdds/dds/topic/TypeSupport.hpp>
+
+#include "alias_struct/gen/alias_struct.hpp"
+#include "alias_struct/gen/alias_structPubSubTypes.hpp"
+
+#include "array_struct/gen/array_struct.hpp"
+#include "array_struct/gen/array_structPubSubTypes.hpp"
+
+#include "bitmask_struct/gen/bitmask_struct.hpp"
+#include "bitmask_struct/gen/bitmask_structPubSubTypes.hpp"
+
+#include "bitset_struct/gen/bitset_struct.hpp"
+#include "bitset_struct/gen/bitset_structPubSubTypes.hpp"
+
+#include "enum_struct/gen/enum_struct.hpp"
+#include "enum_struct/gen/enum_structPubSubTypes.hpp"
+
+#include "extensibility_struct/gen/extensibility_struct.hpp"
+#include "extensibility_struct/gen/extensibility_structPubSubTypes.hpp"
+
+#include "key_struct/gen/key_struct.hpp"
+#include "key_struct/gen/key_structPubSubTypes.hpp"
+
+#include "map_struct/gen/map_struct.hpp"
+#include "map_struct/gen/map_structPubSubTypes.hpp"
+
+#include "primitives_struct/gen/primitives_struct.hpp"
+#include "primitives_struct/gen/primitives_structPubSubTypes.hpp"
+
+#include "sequence_struct/gen/sequence_struct.hpp"
+#include "sequence_struct/gen/sequence_structPubSubTypes.hpp"
+
+#include "string_struct/gen/string_struct.hpp"
+#include "string_struct/gen/string_structPubSubTypes.hpp"
+
+#include "struct_struct/gen/struct_struct.hpp"
+#include "struct_struct/gen/struct_structPubSubTypes.hpp"
+
+#include "union_struct/gen/union_struct.hpp"
+#include "union_struct/gen/union_structPubSubTypes.hpp"
+
+namespace test {
+
+namespace SupportedTypes {
+
+const std::string ALIAS_STRUCT{"alias_struct"};
+const std::string ARRAY_STRUCT{"array_struct"};
+const std::string BITMASK_STRUCT{"bitmask_struct"};
+const std::string BITSET_STRUCT{"bitset_struct"};
+const std::string ENUM_STRUCT{"enum_struct"};
+const std::string EXTENSIBILITY_STRUCT{"extensibility_struct"};
+const std::string KEY_STRUCT{"key_struct"};
+const std::string MAP_STRUCT{"map_struct"};
+const std::string PRIMITIVE_STRUCT{"primitives_struct"};
+const std::string SEQUENCE_STRUCT{"sequence_struct"};
+const std::string STRING_STRUCT{"string_struct"};
+const std::string STRUCT_STRUCT{"struct_struct"};
+const std::string UNION_STRUCT{"union_struct"};
+
+} // namespace SupportedTypes
+
+const std::vector<std::string> supported_types = {
+    SupportedTypes::ALIAS_STRUCT,
+    SupportedTypes::ARRAY_STRUCT,
+    SupportedTypes::BITMASK_STRUCT,
+    SupportedTypes::BITSET_STRUCT,
+    SupportedTypes::ENUM_STRUCT,
+    SupportedTypes::EXTENSIBILITY_STRUCT,
+    SupportedTypes::KEY_STRUCT,
+    SupportedTypes::MAP_STRUCT,
+    SupportedTypes::PRIMITIVE_STRUCT,
+    SupportedTypes::SEQUENCE_STRUCT,
+    SupportedTypes::STRING_STRUCT,
+    SupportedTypes::STRUCT_STRUCT,
+    SupportedTypes::UNION_STRUCT
+};
+
+void register_type_object_representation(
+        const std::string& type_name)
+{
+    using namespace eprosima::fastdds::dds;
+
+    if (type_name == SupportedTypes::ALIAS_STRUCT)
```

Comment:
This must hurt for a switch-lover. You could maybe use an enumeration and define a to_string method. Ignore this if you will.

Reply:
I had that at the beginning. It's the most logical approach. The problem was that I wasn't able to display with GoogleTest the `to_string`  name of the tests, so I kept getting:
```
The following tests passed:
        DynTypeIDLTests/DynTypeIDLTests.to_idl/0
        DynTypeIDLTests/DynTypeIDLTests.to_idl/1
        ...
```
instead of
```
The following tests passed:
        DynTypeIDLTests/DynTypeIDLTests.to_idl/"alias_struct"
        DynTypeIDLTests/DynTypeIDLTests.to_idl/"array_struct"
        ...
```